odl/space/base_ntuples.py

```diff
@@ -403,6 +403,281 @@ def __repr__(self):
+        kwargs:
+            Keyword arguments to ``ufunc.method``.
+
+        Returns
+        -------
+        ufunc_result : # TODO fill in
+            Result of the ufunc evaluation.
+
+        Examples
+        --------
+        We apply `numpy.add` to ODL vectors, also with the optional
+        ``out`` parameter:
+
+        >>> rn = odl.rn(3)
+        >>> x = rn.element([1, 2, 3])
+        >>> y = rn.element([-1, -2, -3])
+        >>> out = rn.element()
+        >>> np.add(x, y)
+        rn(3).element([0.0, 0.0, 0.0])
+        >>> result = np.add(x, y, out=out)
+        >>> out
+        rn(3).element([0.0, 0.0, 0.0])
+        >>> result is out
+        True
+
+        The ``add.accumulate`` method retains the original shape and
+        ``dtype``. We can override the latter with the ``dtype``
+        parameter:
+
+        >>> x = rn.element([1, 2, 3])
+        >>> np.add.accumulate(x)
+        rn(3).element([1.0, 3.0, 6.0])
+        >>> np.add.accumulate(x, dtype=complex)
+        cn(3).element([(1+0j), (3+0j), (6+0j)])
+
+        The ``add.at`` method operates in-place. Here we add the second
+        operand ``[5, 10]`` to ``x`` at indices ``[0, 2]``:
+
+        >>> x = rn.element([1, 2, 3])
+        >>> np.add.at(x, [0, 2], [5, 10])
+        >>> x
+        rn(3).element([6.0, 2.0, 13.0])
+
+        The output of ``add.outer`` would have ``ndim = 2`` axes, so we
+        need to flatten internally:
+
+        >>> x = rn.element([1, 2, 3])
+        >>> y = rn.element([-1, -2, -3])
+        >>> np.add.outer(x, y)
+        rn(9).element([0.0, -1.0, -2.0, ..., 2.0, 1.0, 0.0])
+
+        Using ``add.reduce`` produces a scalar, which can be avoided by
+        using ``keepdims=True``:
+
+        >>> x = rn.element([1, 2, 3])
+        >>> np.add.reduce(x)
+        6.0
+        >>> np.add.reduce(x, keepdims=True)
+        rn(1).element([6.0])
+
+        Finally, ``add.reduceat`` is a combination of ``reduce`` and
+        ``at`` with rather flexible and complex semantics (see the
+        `reduceat documentation`_ for details):
+
+        >>> np.add.reduceat(x, [0, 1])
+        rn(2).element([1.0, 5.0])
+
+        References
+        ----------
+        .. _corresponding NEP:
+           https://github.com/numpy/numpy/blob/master/doc/neps/\
+ufunc-overrides.rst
+
+        .. _interface documentation:
+           https://github.com/charris/numpy/blob/master/doc/source/reference/\
+arrays.classes.rst#special-attributes-and-methods
+
+        .. _general documentation on Numpy ufuncs:
+           https://docs.scipy.org/doc/numpy/reference/ufuncs.html
+
+        .. _reduceat documentation:
+           https://docs.scipy.org/doc/numpy/reference/generated/\
+numpy.ufunc.reduceat.html
+        """
+        # Unwrap out if provided. The output parameters are all wrapped
+        # in one tuple, also if there is only one.
+        out_tuple = kwargs.pop('out', ())
+
+        # We allow our own vectors and `numpy.ndarray` objects as `out`
```

Comment:
Any comment on this? I thought it'd be nice to allow Numpy arrays as `out` but I'm not 100 % about it.